src/services/LoggingService.js

```diff
@@ -3,36 +3,67 @@
  *
  * Requires the NewRelic Browser JavaScript snippet.
  */
+
+// NewRelic will not log an error if it is too long.
+const MAX_ERROR_LENGTH = 4000;
+
+function fixErrorLength(error) {
+  if (error.message && error.message.length > MAX_ERROR_LENGTH) {
+    const processedError = Object.create(error);
+    processedError.message = processedError.message.substring(0, MAX_ERROR_LENGTH);
+    return processedError;
+  } else if (typeof error === 'string' && error.length > MAX_ERROR_LENGTH) {
+    return error.substring(0, MAX_ERROR_LENGTH);
+  }
+  return error;
+}
+
 class LoggingService {
   static logInfo(message) {
     if (typeof newrelic !== 'undefined') {
       newrelic.addPageAction('INFO', { message });
     }
   }
 
-  static logError(error) {
+  static logError(error, customAttributes) {
     if (typeof newrelic !== 'undefined') {
       // Note: customProperties are not sent.  Presumably High-Security Mode is being used.
-      newrelic.noticeError(error);
+      newrelic.noticeError(fixErrorLength(error), customAttributes);
```

Comment:
@abutterworth: FYI: The problem we were having with custom attributes was that it is a newer feature and wasn't supported by the version of the newrelic snippet we had installed.  The npm package update brings in a newer snippet that supports these custom attributes.